### DISCUSSION TITLE
Can Zustand be used without Node/npm?

### DISCUSSION THREAD

**Discussion by electrotype** [pmndrs/zustand#1443] [General] [answered]
My application is not based on Node and doesn't use React everywhere. I'm starting to use React only on pages with a lot of javascript.

In my application, I simply added a link to those .js files:

- react.production.min.js
- react-dom.production.min.js

And it works perfectly. 

But I'd like to try Zustand. Can Zustand be used without Node/npm too? What files should be loaded? Is there an example somewhere?

**Comment by dai-shi** [comment, marked as answer]
https://unpkg.com/browse/zustand@4.1.4/umd/
You can try umd bundles. Not many users do this, so there might be some issues.

Oh no, `use-sync-external-store` package doesn't provide UMD.

It works with `use-zustand` which doesn't use `use-sync-external-store`.
See: https://jsfiddle.net/ugjznf90/10/

<details>
<summary>full code</summary>

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <title>Hello World</title>
    <script src="https://unpkg.com/react@18/umd/react.development.js"></script>
    <script src="https://unpkg.com/react-dom@18/umd/react-dom.development.js"></script>
    <script src="https://unpkg.com/zustand@4.1.4/umd/vanilla.development.js"></script>
    <script src="https://unpkg.com/use-zustand@0.0.1/dist/index.umd.js"></script>

    <!-- Don't use this in production: -->
    <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
  </head>
  <body>
    <div id="root"></div>
    <script type="text/babel">
      const store = zustandVanilla((set) => ({
        count: 0,
        inc: () => set((state) => ({ count: state.count + 1 })),
      }));
      function MyApp() {
        const { count, inc } = useZustand.useZustand(store, (state) => state);
        return (
          <div>
            <h1>Hello, world!</h1>
            {count} <button onClick={inc}>+1</button>
          </div>
        )
      }

      const container = document.getElementById("root");
      const root = ReactDOM.createRoot(container);
      root.render(<MyApp />);
    </script>
    <!--
      Note: this page is a great way to try React but it's not suitable for production.
      It slowly compiles JSX with Babel in the browser and uses a large development build of React.

      Read this section for a production-ready setup with JSX:
      https://reactjs.org/docs/add-react-to-a-website.html#add-jsx-to-a-project

      In a larger project, you can use an integrated toolchain that includes JSX instead:
      https://reactjs.org/docs/create-a-new-react-app.html

      You can also use React without JSX, in which case you can remove Babel:
      https://reactjs.org/docs/react-without-jsx.html
    -->
  </body>
</html>
```
</details>


based on: https://github.com/reactjs/reactjs.org/tree/main/static/html

**Comment by electrotype** [comment]
Thanks!!

What is `use-zustand`? Where is the source code? 

Should using the `0.0.1` version of this package/file ok with any future Zustand version?

Also, is this really all Zustand's code? https://unpkg.com/zustand@4.1.4/umd/vanilla.development.js
47 lines and that's it? 

Thanks for the help!

**Reply by dai-shi** [reply]
use-zustand is my experimental library to use zustand store without useSyncExternalStore. https://github.com/dai-shi/use-zustand

To use the official zustand hook https://unpkg.com/browse/zustand@4.1.4/umd/index.development.js, you need to ask use-sync-external-store to provide UMD.

Yes, zustand is very small. Here's vanilla + hook: https://twitter.com/dai_shi/status/1583082766081531905
